### DISCUSSION TITLE
Formulae `gspell` is missing `icu-uc` dependency on OSX

### DISCUSSION THREAD

**Discussion by sp1ritCS** [Homebrew/discussions#3538] [Everyday usage] [answered]
*this probably warrants an issue on [Homebrew/homebrew-core](https://github.com/Homebrew/homebrew-core), but I don't have a mac to test it outside the container*

I have a [GH CI run](https://github.com/sp1ritCS/notekit/blob/b33d56037ab2e0e5d9bae7d04a9b00be42b9b80d/.github/workflows/ci.yml#L73), that uses homebrew to download it's dependencies on OSX and Linux, gspell-1 being one of them.

The run works flawlessly on Linux, but the meson invocation fails on OSX with:
```
Run-time dependency gspell-1 found: NO (tried framework and cmake)

meson.build:20:0: ERROR: Could not generate cargs for gspell-1:
Package icu-uc was not found in the pkg-config search path.
Perhaps you should add the directory containing `icu-uc.pc'
to the PKG_CONFIG_PATH environment variable
Package 'icu-uc', required by 'gspell-1', not found
```
(see: https://github.com/sp1ritCS/notekit/runs/7855131333?check_suite_focus=true)

since `gspell-1` requires icu-ic ([gspell/configure.ac#95](https://gitlab.gnome.org/GNOME/gspell/-/blob/cd76632d673bc81fca2c484204644cf78d936f6b/configure.ac#L95)) and it doesn't appear to be listed in the [formulae of gspell](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gspell.rb) at all, I suspect icu should be added to it.

I believe this isn't an issue on Linux, as another dependency is seemingly pulling icu in (but only on linux, not osx), so gspell just uses this.

**Comment by SMillerDev** [comment]
ICU4C is provided by macOS so it probably uses that. Which explains why the build succeeds but there is no pkg-config file.

**Reply by sp1ritCS** [reply]
this is probably the case. I assume it links against the system library on osx (nothing pulls brews icu depdendency otherwise).

so the formulae should modify it's pkg-config file to drop it's icu-uc requirement and instead add libicu as linker flag itself on osx, shouldn't it?

**Reply by carlocab** [reply]
> so the formulae should modify it's pkg-config file to drop it's icu-uc requirement and instead add libicu as linker flag itself on osx, shouldn't it?

`icu-uc.pc` provides `libicuuc` and `libicudata` while macOS provides `libicucore`, so it's not clear that doing this is appropriate.

In any case, `gspell` already has an indirect dependency on `icu4c` through `gtk+3` (via `pango` and `harfbuzz`).

**Comment by carlocab** [comment, marked as answer]
`gspell` has a dependency on `icu4c`:
```
❯ brew deps gspell | grep -q icu4c && echo 'has icu4c'
has icu4c
```
I think the issue you're having is that `icu4c` is keg-only:
```
❯ brew info icu4c
==> icu4c: stable 70.1 (bottled) [keg-only]
[snip]
==> Caveats
icu4c is keg-only, which means it was not symlinked into /usr/local,
because macOS provides libicucore.dylib (but nothing else).

If you need to have icu4c first in your PATH, run:
  echo 'export PATH="/usr/local/opt/icu4c/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/usr/local/opt/icu4c/sbin:$PATH"' >> ~/.zshrc

For compilers to find icu4c you may need to set:
  export LDFLAGS="-L/usr/local/opt/icu4c/lib"
  export CPPFLAGS="-I/usr/local/opt/icu4c/include"

For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig"
```
You probably need, in particular, the recommendation about `PKG_CONFIG_PATH`.

**Reply by sp1ritCS** [reply]
ok, I went this route. No idea if it's giving me the aame headers of the library it was linked against, but it seems to work.

But shouldn't gspells pkg-config still be modified to make this the default behaviour? having to set PKG_CONFIG_PATH every time doesn't seem to be the way to go.

**Reply by carlocab** [reply]
`brew linkage gspell` reports

```
System libraries:
  /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
  /System/Library/Frameworks/Carbon.framework/Versions/A/Carbon
  /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa
  /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
  /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
  /usr/lib/libSystem.B.dylib
  /usr/lib/libobjc.A.dylib
Homebrew libraries:
  /usr/local/opt/atk/lib/libatk-1.0.0.dylib (atk)
  /usr/local/opt/cairo/lib/libcairo-gobject.2.dylib (cairo)
  /usr/local/opt/cairo/lib/libcairo.2.dylib (cairo)
  /usr/local/opt/enchant/lib/libenchant-2.2.dylib (enchant)
  /usr/local/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (gdk-pixbuf)
  /usr/local/opt/gettext/lib/libintl.8.dylib (gettext)
  /usr/local/opt/glib/lib/libgio-2.0.0.dylib (glib)
  /usr/local/opt/glib/lib/libglib-2.0.0.dylib (glib)
  /usr/local/opt/glib/lib/libgobject-2.0.0.dylib (glib)
  /usr/local/Cellar/gspell/1.10.0/lib/libgspell-1.2.dylib (gspell)
  /usr/local/opt/gtk+3/lib/libgdk-3.0.dylib (gtk+3)
  /usr/local/opt/gtk+3/lib/libgtk-3.0.dylib (gtk+3)
  /usr/local/opt/gtk-mac-integration/lib/libgtkmacintegration-gtk3.4.dylib (gtk-mac-integration)
  /usr/local/opt/harfbuzz/lib/libharfbuzz.0.dylib (harfbuzz)
  /usr/local/opt/icu4c/lib/libicudata.70.dylib (icu4c)
  /usr/local/opt/icu4c/lib/libicuuc.70.dylib (icu4c)
  /usr/local/opt/pango/lib/libpango-1.0.0.dylib (pango)
  /usr/local/opt/pango/lib/libpangocairo-1.0.0.dylib (pango)
```
so you're definitely getting the same library that it was linked against.

> But shouldn't gspells pkg-config still be modified to make this the default behaviour? having to set PKG_CONFIG_PATH every time doesn't seem to be the way to go.

Well, if you know of a way to do this, then we're happy to review a pull request that does it.
